internal/services/cognitive/cognitive_account_data_source.go

```diff
@@ -34,56 +36,248 @@ func dataSourceCognitiveAccount() *pluginsdk.Resource {
+										Computed: true,
+									},
+
+									"ignore_missing_vnet_service_endpoint": {
+										Type:     pluginsdk.TypeBool,
+										Computed: true,
+									},
+								},
+							},
+						},
+
+						"bypass": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"network_injection": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"scenario": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"subnet_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"outbound_network_access_restricted": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
 			"primary_access_key": {
 				Type:      pluginsdk.TypeString,
 				Computed:  true,
 				Sensitive: true,
 			},
 
+			"project_management_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
+			"public_network_access_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
+			"qna_runtime_endpoint": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
 			"secondary_access_key": {
 				Type:      pluginsdk.TypeString,
 				Computed:  true,
 				Sensitive: true,
 			},
 
-			"identity": commonschema.SystemAssignedUserAssignedIdentityComputed(),
+			"sku_name": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"storage": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"storage_account_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"identity_client_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
 
 			"tags": commonschema.Tags(),
 		},
 	}
 }
 
+func flattenCognitiveAccountDataSourceNetworkAcls(input *cognitiveservicesaccounts.NetworkRuleSet) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	ipRules := make([]interface{}, 0)
+	if input.IPRules != nil {
+		for _, v := range *input.IPRules {
+			ipRules = append(ipRules, v.Value)
+		}
+	}
+
+	virtualNetworkRules := make([]interface{}, 0)
+	if input.VirtualNetworkRules != nil {
+		for _, v := range *input.VirtualNetworkRules {
+			id := v.Id
+			subnetId, err := commonids.ParseSubnetIDInsensitively(v.Id)
+			if err == nil {
+				id = subnetId.ID()
+			}
+
+			virtualNetworkRules = append(virtualNetworkRules, map[string]interface{}{
+				"subnet_id":                            id,
+				"ignore_missing_vnet_service_endpoint": *v.IgnoreMissingVnetServiceEndpoint,
+			})
+		}
+	}
+
+	return []interface{}{map[string]interface{}{
+		"bypass":                input.Bypass,
+		"default_action":        input.DefaultAction,
+		"ip_rules":              ipRules,
+		"virtual_network_rules": virtualNetworkRules,
+	}}
+}
```

Comment:
This needs to be moved to be below the `dataSourceCognitiveAccountRead` function to follow the resource structure standards, the order should be:
- Resource Definition
- CRUD
- Helper Functions (flatten/expand)